powerline/segments/i3wm.py

```diff
@@ -16,12 +25,49 @@ def calcgrp(w):
 	return group
 
 
-def workspaces(pl):
-	'''Return workspace list
+def workspaces(pl, strip=0):
+	'''Return list of used workspaces
+
+	:param int strip:
+		Specifies how many characters from the front of each workspace name should
+		be stripped (e.g. to remove workspace numbers). Defaults to zero.
 
 	Highlight groups used: ``workspace``, ``w_visible``, ``w_focused``, ``w_urgent``
 	'''
+	global conn
+	if not conn: conn = i3ipc.Connection()
+
 	return [{
-		'contents': w['name'],
+		'contents': w['name'][min(len(w['name']),strip):],
 		'highlight_groups': calcgrp(w)
-	} for w in i3.get_workspaces()]
+	} for w in conn.get_workspaces()]
+
+class ModeSegment(Segment):
+	argmethods = ('render')
```

Comment:
This is the same as `argmethods = ('r', 'e', 'n', 'd', 'e', 'r')`. Also I do not see where this method is defined.